src/launchpad/components/Console.js

```diff
@@ -13,44 +17,38 @@ const Console = React.createClass({
 
   displayName: "Console",
 
-  getInitialState: function() {
-    return {
-      expressions: []
-    };
-  },
-
-  evaluate(expression) {
-    this.props.client.clientCommands.evaluate(expression, {})
-      .then(
-        result => this.addExpression(expression, result),
-        err => console.warn("Error when evaluating expression", err)
-      );
-  },
-
-  addExpression(expression, result) {
-    this.setState(prevState => ({
-      expressions: [{
-        input: expression,
-        packet: result,
-      }, ...prevState.expressions]
-    }));
-  },
-
-  clearResultsList() {
-    this.setState(prevState => ({
-      expressions: []
-    }));
-  },
-
   render: function() {
+    let {
+      actions,
+      expressions,
+    } = this.props;
+
     return dom.main({},
       Header({
-        evaluate: this.evaluate,
-        clearResultsList: this.clearResultsList
+        evaluate: expression => actions.evaluateInput(
+          this.props.client.clientCommands.evaluate,
```

Comment:
It'd be nice to move this to the action. Checkout how the debugger passss client in through makeArgs. It's the best thing we did with the redux architecture. The big win is that only the actions have the client.